ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpersImpl.java

```diff
@@ -87,10 +89,18 @@
 
     private Ds3ClientHelpers.Job innerStartReadJob(final String bucket, final Iterable<Ds3Object> objectsToRead, final ReadJobOptions options)
             throws SignatureException, IOException, XmlProcessingException {
-        final BulkGetResponse prime = this.client.bulkGet(new BulkGetRequest(bucket, Lists.newArrayList(objectsToRead))
+        final List<Ds3Object> objects = Lists.newArrayList(objectsToRead);
+        final BulkGetResponse prime = this.client.bulkGet(new BulkGetRequest(bucket, objects)
                 .withChunkOrdering(ChunkClientProcessingOrderGuarantee.NONE)
                 .withPriority(options.getPriority()));
-        return new ReadJobImpl(this.client, prime.getResult());
+
+        final ImmutableMultimap<String, Range> partialRanges = PartialObjectHelpers.getPartialObjectsRanges(objects);
+
+        final MasterObjectList mob = prime.getResult();
+
+        final ImmutableMultimap<BulkObject, Range> blobtoRanges = PartialObjectHelpers.mapRangesToBlob(mob.getObjects(), partialRanges);
```

Comment:
This is changed in my local repo.  I had to pass in the multi map instead of the list